src/Aspire.Hosting/ApplicationModel/ReferenceExpression.cs

```diff
@@ -149,6 +153,41 @@ public void AppendFormatted<T>(IResourceBuilder<T> valueProvider)
 
         internal readonly ReferenceExpression GetExpression() =>
             new(_builder.ToString(), [.. _valueProviders], [.. _manifestExpressions]);
+
+        private static string EscapeUnescapedBraces(string input)
```

Comment:
Should this assume `{{` is already escaped or just always escape `{` (and closing)? It's a breaking change either way...